src/store/state.ts

```diff
@@ -148,13 +149,21 @@ function firstNonSelectedIndex(state: VQBState) {
  * @return the subpart of the pipeline that is currently active.
  */
 export function activePipeline(state: VQBState) {
-  return currentPipeline(state).slice(0, firstNonSelectedIndex(state));
+  const pipeline = currentPipeline(state);
+  if (!pipeline) {
+    return;
+  }
+  return pipeline.slice(0, firstNonSelectedIndex(pipeline, state.selectedStepIndex));
```

Comment:
You could probably use the optional call construct and remove the `if` (untested):

```suggestion
  return pipeline?.slice(0, firstNonSelectedIndex(pipeline, state.selectedStepIndex));
```